From: riel@redhat.com
Subject: Various bits to support 2.6.11-rc1

Some tidbits to make Xen (-unstable) compile with the latest
2.6 snapshot.  No need for you to duplicate the work I already
did.

Rediffed and 
Acked-by: garloff@suse.de
Signed-off-by: Rik van Riel <riel@redhat.com>

Index: linux-2.6.10/arch/xen/i386/Kconfig
===================================================================
--- linux-2.6.10.orig/arch/xen/i386/Kconfig	2005-02-22 12:44:19.000000000 +0100
+++ linux-2.6.10/arch/xen/i386/Kconfig	2005-02-22 12:55:57.000000000 +0100
@@ -253,6 +253,10 @@
 	depends on !M386
 	default y
 
+config GENERIC_CALIBRATE_DELAY
+	bool
+	default y
+
 config X86_PPRO_FENCE
 	bool
 	depends on M686 || M586MMX || M586TSC || M586 || M486 || M386
Index: linux-2.6.10/arch/xen/i386/kernel/i386_ksyms.c
===================================================================
--- linux-2.6.10.orig/arch/xen/i386/kernel/i386_ksyms.c	2005-02-22 12:44:19.000000000 +0100
+++ linux-2.6.10/arch/xen/i386/kernel/i386_ksyms.c	2005-02-22 12:55:57.000000000 +0100
@@ -61,7 +61,6 @@
 
 /* platform dependent support */
 EXPORT_SYMBOL(boot_cpu_data);
-EXPORT_SYMBOL(MCA_bus);
 #ifdef CONFIG_DISCONTIGMEM
 EXPORT_SYMBOL(node_data);
 EXPORT_SYMBOL(physnode_map);
Index: linux-2.6.10/kernel/sysctl.c
===================================================================
--- linux-2.6.10.orig/kernel/sysctl.c	2005-02-22 12:41:22.000000000 +0100
+++ linux-2.6.10/kernel/sysctl.c	2005-02-22 12:55:57.000000000 +0100
@@ -653,7 +653,7 @@
 		.proc_handler   = &proc_unknown_nmi_panic,
 	},
 #endif
-#if defined(CONFIG_X86)
+#if defined(CONFIG_X86) && !defined(CONFIG_XEN)
 	{
 		.ctl_name	= KERN_BOOTLOADER_TYPE,
 		.procname	= "bootloader_type",
